### PR TITLE
Fix: --initial-flag only available on recent git versions

### PR DESCRIPTION
**Pull request by rsicart**

### CODE REVIEW

File: hooks/post_gen_project.sh [hunk 1]
```diff
@@ -1,5 +1,14 @@
 #!/usr/bin/env bash
 if [[ '{{ cookiecutter.init_repository }}' == 'y' ]]; then
-    git init --initial-branch=main
+    version_major=$(git --version | awk '{print $3}' | awk -F'.' '{print $1}')
+    version_minor=$(git --version | awk '{print $3}' | awk -F'.' '{print $2}')
+    version_major=$((version_major + 0))
+    version_minor=$((version_minor + 0))
+    if [[ $version_major -eq 2 && $version_minor -lt 28 ]]; then
+       git init
+       git co -b main
```
Comment (ElieDeloumeau) [inline hooks/post_gen_project.sh:9]: Please do not use aliases

```suggestion
       git checkout -b main
```

**Comment by ElieDeloumeau** [comment]
Hi @rsicart, thank you for your PR 🙂

**Comment by xp-1000** [comment]
wouldn't it be easier to always run git init && git checkout -b main no matter the version?

File: hooks/post_gen_project.sh [hunk 2]
```diff
@@ -1,5 +1,14 @@
 #!/usr/bin/env bash
 if [[ '{{ cookiecutter.init_repository }}' == 'y' ]]; then
-    git init --initial-branch=main
+    version_major=$(git --version | awk '{print $3}' | awk -F'.' '{print $1}')
+    version_minor=$(git --version | awk '{print $3}' | awk -F'.' '{print $2}')
+    version_major=$((version_major + 0))
+    version_minor=$((version_minor + 0))
```
Comment (xp-1000) [inline hooks/post_gen_project.sh:6]:
```suggestion
IFS='.' read version_major version_middle version_minor <<< $(git --version | awk '{print $3}')
```

File: hooks/post_gen_project.sh (hunk 1 above)
Comment (rsicart) [reply hooks/post_gen_project.sh:9]: :man_facepalming:

**Comment by rsicart** [comment]
> wouldn't it be easier to always run git init && git checkout -b main no matter the version?

I agree

**Pull request merged**